counterpartylib/lib/messages/versions/enhanced_send.py

```diff
@@ -0,0 +1,202 @@
+#! /usr/bin/python3
+
+import struct
+import json
+import logging
+logger = logging.getLogger(__name__)
+
+from counterpartylib.lib import (config, util, exceptions, util, message_type, address)
+
+FORMAT = '>QQ21s'
+LENGTH = 8 + 8 + 21
+MAX_MEMO_LENGTH = 34
+ID = 2 # 0x02
+
+def unpack(db, message, block_index):
+    try:
+        # account for memo bytes
+        memo_bytes_length = len(message) - LENGTH
+        if memo_bytes_length < 0:
+            raise exceptions.UnpackError('invalid message length')
+        if memo_bytes_length > MAX_MEMO_LENGTH:
+            raise exceptions.UnpackError('memo too long')
+
+        struct_format = FORMAT + ('{}s'.format(memo_bytes_length))
+        asset_id, quantity, short_address_bytes, memo_bytes = struct.unpack(struct_format, message)
+        if len(memo_bytes) == 0:
+            memo_bytes = None
+        
+        # unpack address
+        try:
+          full_address = address.unpack(short_address_bytes)
```

Comment:
Turns out there is no possible failure case.  Any 21 bytes are valid and `short_address_bytes` must be 21 bytes long if we got here.  I removed this check.